HOW/tech/applications/cell-joining-code/cell-joining-code-v1.md

```diff
@@ -0,0 +1,110 @@
+---
+HOW: /tech/applications/cell-joining-code
+Source: /social/process/how/how-v1
+Status: Draft
+Created: 2024-11-23
+Version: idx/1
+---
+
+## Title
+
+Cell Joining Code
+
+## Authors
+
+@mattyg <matt@buildyourweb.app>, @zippy <eric@holochain.org>, Claude AI 3.5 Sonnet
+
+## Abstract
+
+This HOW proposes a standardized URI format for cell joining codes that enables conductors to join specific networks. The format encapsulates all necessary information for network joining in a single, shareable URI string, excluding the actual DNA.  The format also allows app-specific data to be carried so that UIs can display arbitrary data as needed for good 
+
+## Motivation
+
+Joining a Cell network requires multiple pieces of information to be shared: DNA hash, network seed, membrane proof, and app-specific metadata. Several apps have run into the need to share this information as a single string and implemented their own joining code format.
+
+A standardized joining code format would:
+- Simplify the network joining process
+- Simplify use of network joining codes in apps
+- Reduce user error during configuration
+- Enable one-click joining capabilities
+- Facilitate easier sharing of network access
+- Facilitate services providing "always-on" Cells: simply provide the DNA and joining code
+
+## Rationale
+
+This specification meets the criteria of the motivation because:
+- it uses the standard URI format which makes it amenable to deep-linking
+- it uses a specific path in the URI format leaving other paths available under the `hc://` scheme for other uses.
+- it includes a well-known pattern for specifying attributes and thus arbitrary application-specific components, as well as optional data elements.
+
+## Technical Specification
+
+The cell joining URI format MUST follow this structure:
+
+```
+hc://join/<dna_hash>/<dna_modifiers>?proof=<membrane_proof>&metadata=<metadata>
+```
+
+Where:
+
+- The scheme MUST be `hc://`
```

Comment:
```suggestion
- The prefix MUST be `hc:`
```

I'm thinking this should not be an actual URI, because if you did want to use deep-linking in your app, you would want your own personal scheme for that app. Otherwise, if you supported `hc://` deep links for your app, then joining codes intended for other apps might still be opened by yours.

On the other hand, if we keep this a URI, then we assume it will be used for deep linking, an app can just refuse to use a joining code not intended for itself. Would it be invalid to have a uri with multiple schemes? I.e. `my-app://hc://join/...`

Reply:
> Is it valid uri format to have a uri with multiple schemes? I.e. my-app://hc://join/.... If so that might give us the best of both worlds: create an app-specific scheme if you want, otherwise use the generic one. If this is not a valid uri, then I'm leaning towards not having it be a uri at all.

No it is not valid to have multiple schemes in a URI. Thus I think the best approach is to make this just a path, not a URI.